src/lib.rs

```diff
@@ -19,33 +20,126 @@ mod serialisation;
 pub use self::{
     errors::{Error, Result},
     location::{Aggregation, DstLocation, EndUser, Itinerary, SrcLocation},
+    msg::Msg,
     msg_id::MessageId,
     serialisation::WireMsg,
 };
 use bytes::Bytes;
+use threshold_crypto::PublicKey;
+use xor_name::XorName;
 
 /// Type of message.
 /// Note this is part of this crate's public API but this enum is
 /// never serialised or even part of the message that is sent over the wire.
-#[derive(PartialEq, Debug)]
+#[derive(PartialEq, Debug, Clone)]
```

Comment:
We need to be careful deriving Clone on types that can be very large. IMO it's best if we avoid this as much as possible so we are naturally careful with `.clone()`.